File: program/program/find-the-largest-two-elements-in-an-array/find_the_largest_two_elements_in_an_array.go
package main

import (
    "fmt"
    "sort"
)

func main() {
    fmt.Println(largestTwoElements([]int{12, 13, 1, 10, 34, 1}))
    fmt.Println(largestTwoElements([]int{10, 5, 10}))
    fmt.Println(largestTwoElements([]int{10, 10, 10}))
}

func largestTwoElements(arr []int) []int {
    // sort the input array in descending order
    sort.Sort(sort.Reverse(sort.IntSlice(arr)))

    // return the first 2 elements
    return arr[:2]
}

File: program/find-the-largest-two-elements-in-an-array/FindTheLargestTwoElementsInAnArray.js
function findLargestTwoElements(arr) {
    if (arr.length < 2) {
      console.log("Array should have at least two elements.");
      return;
    }
  
    // Sort the array in descending order
    arr.sort(function(a, b) {
      return b - a;
    });
  
    // Extract the first and second elements (largest and second largest)
    let largest = arr[0];
    let secondLargest = arr[1];
  
    return [largest, secondLargest];
  }

    // Example usage
    let numbers = [10, 5, 8, 20, 15];
    let result = findLargestTwoElements(numbers);

    let letters = ['f', 'o', 'p', 'w', 'z'];
    let result = findLargestTwoElements(letters);
    
    console.log("Largest two numbers:", result);